make/tests

```diff
@@ -1,3 +1,59 @@
+ifdef STANC2
+TEST_STANC=test/test-models/stanc2$(EXE)
+else
+TEST_STANC=bin/stanc$(EXE)
```

Comment:
The stanc3 binary is placed in the bin folder, the existing one remains in the test-models folder. I checked that make clean removes it. Running tests with stanc2=true will use the old compiler for everything.